py/test/selenium/webdriver/marionette/mn_options_tests.py

```diff
@@ -15,25 +15,114 @@
 # specific language governing permissions and limitations
 # under the License.
 
-
+import errno
 import pytest
+import shutil
+import tempfile
+import types
 
+from selenium.webdriver.firefox.firefox_binary import FirefoxBinary
+from selenium.webdriver.firefox.firefox_profile import FirefoxProfile
+from selenium.webdriver.firefox.options import Log, Options
 from selenium.webdriver import Firefox
-from selenium.webdriver.firefox.options import Options
 
 
 @pytest.fixture
-def driver(capabilities):
-    options = Options()
-    driver = Firefox(
-        capabilities=capabilities,
-        firefox_options=options)
+def driver(options = None):
+    if options is None:
+        options = Options()
+    driver = Firefox(firefox_options=options)
     yield driver
     driver.quit()
 
 
-class TestOptions(object):
-
+class TestIntegration(object):
     def test_we_can_pass_options(self, driver, pages):
-        pages.load('formPage.html')
+        pages.load("formPage.html")
         driver.find_element_by_id("cheese")
+
+
+class TestUnit(object):
+    def test_ctor(self):
+        opts = Options()
+        assert opts._binary is None
+        assert opts._profile is None
+        assert opts._arguments == []
+        assert isinstance(opts.log, Log)
+
+    def test_binary(self):
+        opts = Options()
+        assert opts.binary is None
+
+        other_binary = FirefoxBinary()
+        assert other_binary != opts.binary
+        opts.binary = other_binary
+        assert other_binary == opts.binary
+
+        path = "/path/to/binary"
+        opts.binary = path
+        assert isinstance(opts.binary, FirefoxBinary)
+        assert opts.binary._start_cmd == path
+
+    def test_profile(self):
+        opts = Options()
+        assert opts.profile is None
+
+        other_profile = FirefoxProfile()
+        assert other_profile != opts.profile
+        opts.profile = other_profile
+        assert other_profile == opts.profile
+
+        with TemporaryDirectory() as path:
```

Comment:
You should be able to use the built-in pytest fixture `tmpdir` instead: http://docs.pytest.org/en/latest/tmpdir.html